lib/public/utilities/mergeRemoteData.js

```diff
@@ -0,0 +1,74 @@
+/**
+ *  @license
+ *  Copyright CERN and copyright holders of ALICE O2. This software is
+ *  distributed under the terms of the GNU General Public License v3 (GPL
+ *  Version 3), copied verbatim in the file "COPYING".
+ *
+ *  See http://alice-o2.web.cern.ch/license for full licensing information.
+ *
+ *  In applying this license CERN does not waive the privileges and immunities
+ *  granted to it by virtue of its status as an Intergovernmental Organization
+ *  or submit itself to any jurisdiction.
+ */
+
+import { RemoteData } from '/js/src/index.js';
+
+/**
+ * Merge list of remote data to single remote data instance according to following rules
+ * If at least one of them is RemoteData.Failure then RemoteData.Failure is returned, otherwise
+ * If at least one of them is RemoteData.Loading then RemoteData.Loading is returned, otherwise
+ * If at least one of them is RemoteData.NotAsked then RemoteData.NotAsked is returned, otherwise
+ * RemoteData.Success with list of payloads of given remote data list with preserved order is returned
+ *
+ * @param {RemoteData[]} remoteDataList list of remote data
+ * @param {object} [options] addiation options
+ * @param {boolean} [options.flatErrorsList = true] if true errors from Failure remote data are merged to single list
```

Comment:
I think this do not belongs to this. A flatMap or simply flat function should be called right after.

Reply:
refactored